sqlstore/src/main/resources/db/migration_afterMigrate/05_trigger_box.sql

```diff
@@ -0,0 +1,173 @@
+-- StartNoTest
+DELIMITER //
+
+CREATE TRIGGER BoxChange BEFORE UPDATE ON Box
+FOR EACH ROW
+  BEGIN
+  DECLARE log_message varchar(500) CHARACTER SET utf8;
+  SET log_message = CONCAT_WS(', ',
+    CASE WHEN NEW.alias <> OLD.alias THEN CONCAT('alias: ', OLD.alias, ' → ', NEW.alias) END,
+    CASE WHEN (NEW.identificationBarcode IS NULL) <> (OLD.identificationBarcode IS NULL) OR NEW.identificationBarcode <> OLD.identificationBarcode THEN CONCAT('barcode: ', COALESCE(OLD.identificationBarcode, 'n/a'), ' → ', COALESCE(NEW.identificationBarcode, 'n/a')) END,
+    CASE WHEN (NEW.locationBarcode IS NULL) <> (OLD.locationBarcode IS NULL) OR NEW.locationBarcode <> OLD.locationBarcode THEN CONCAT('location: ', COALESCE(OLD.locationBarcode, 'n/a'), ' → ', COALESCE(NEW.locationBarcode, 'n/a')) END,
+    CASE WHEN (NEW.description IS NULL) <> (OLD.description IS NULL) OR NEW.description <> OLD.description THEN CONCAT('description: ', COALESCE(OLD.description, 'n/a'), ' → ', COALESCE(NEW.description, 'n/a')) END
+  );
+  IF log_message IS NOT NULL AND log_message <> '' THEN
+    INSERT INTO BoxChangeLog(boxId, columnsChanged, userId, message) VALUES (
+      New.boxId,
+      COALESCE(CONCAT_WS(',',
+        CASE WHEN NEW.alias <> OLD.alias THEN 'alias' END,
+        CASE WHEN (NEW.identificationBarcode IS NULL) <> (OLD.identificationBarcode IS NULL) OR NEW.identificationBarcode <> OLD.identificationBarcode THEN 'identificationBarcode' END,
+        CASE WHEN (NEW.locationBarcode IS NULL) <> (OLD.locationBarcode IS NULL) OR NEW.locationBarcode <> OLD.locationBarcode THEN 'locationBarcode' END,
+        CASE WHEN (NEW.description IS NULL) <> (OLD.description IS NULL) OR NEW.description <> OLD.description THEN 'description' END), ''),
+      NEW.lastModifier,
+      log_message
+    );
+  END IF;
+  END//
+
+DROP TRIGGER IF EXISTS BoxInsert//
+CREATE TRIGGER BoxInsert AFTER INSERT ON Box
+FOR EACH ROW
+  INSERT INTO BoxChangeLog(boxId, columnsChanged, userId, message) VALUES (
+    NEW.boxId,
+    '',
+    NEW.lastModifier,
+    'Box created.'
+  )//
+
+DELIMITER ;
+-- EndNoTest
+
+DROP VIEW IF EXISTS CompletedPartitions;
+CREATE OR REPLACE VIEW RunPartitionsByHealth AS
```

Comment:
These non-box things (here down to SampleBoxPosition) look to be mis-categoried...